src/nupic/ntypes/Sdr.hpp

```diff
@@ -519,61 +491,124 @@ class SparseDistributedRepresentation : public Serializable
     }
 
     /**
-     * Calculates the number of true / non-zero values in the SDR.  If the SDRs
-     * value is unset this raises an exception.
+     * Calculates the number of true / non-zero values in the SDR.
      *
      * @returns The number of true values in the SDR.
      */
     UInt getSum()
-        { return getFlatIndex().size(); };
+        { return getFlatSparse().size(); };
 
     /**
      * Calculates the sparsity of the SDR, which is the fraction of bits which
      * are true out of the total number of bits in the SDR.
      * I.E.  sparsity = sdr.getSum() / sdr.size
      *
-     * If the SDRs value is unset this raises an exception.
-     *
      * @returns The fraction of values in the SDR which are true.
      */
     Real getSparsity()
         { return (Real) getSum() / size; };
 
     /**
-     * TODO ...
+     * Calculates the number of true bits which both SDRs have in common.
      *
-     * @returns ...
+     * @param sdr, An SDR to compare with, both SDRs must have the same
+     * dimensons.
+     *
+     * @returns Integer, the number of true values which both SDRs have in
+     * common.
      */
     UInt overlap(SparseDistributedRepresentation &sdr) {
-        NTA_ASSERT( false /* Unimplemented */ );
-        return 0;
+        NTA_ASSERT( dimensions.size() == sdr.dimensions.size() );
+        for( UInt i = 0; i < dimensions.size(); i++ )
+            NTA_ASSERT( dimensions[i] == sdr.dimensions[i] );
+
+        UInt ovlp = 0;
+        auto a = this->getDense();
+        auto b = sdr.getDense();
+        for( UInt i = 0; i < size; i++ )
+            ovlp += a[i] && b[i];
+        return ovlp;
     };
 
     /**
      * Make a random SDR, overwriting the current value of the SDR.  The
-     * resulting has uniformly random activations.
+     * result has uniformly random activations.
      *
      * @param sparsity The sparsity of the randomly generated SDR.
+     *
+     * @param rng The random number generator to draw from, optional.
      */
     void randomize(Real sparsity) {
-        NTA_ASSERT( false /* Unimplemented */ );
+        Random rng;
```

Comment:
if this method is used, it'll break determinism in our algos; should introduce a constructor param seed with a default value and use it here for rng